### PR TITLE
Bugfix & Test java.lang.Integer.parseInt

### PR DESCRIPTION
**Pull request by ankitson**
Hi, I ran into this while working on https://github.com/scala-native/scala-native/issues/242. I think parseInt will ignore the first digit if theres no negative sign.

### CODE REVIEW

File: unit-tests/src/main/scala/java/lang/IntegerSuite.scala [hunk 1]
```diff
@@ -0,0 +1,38 @@
+package java.lang
+
+object IntegerSuite extends tests.Suite {
```
Comment (xavier-fernandez) [inline unit-tests/src/main/scala/java/lang/IntegerSuite.scala]: I'm missing some test for the corner cases: `0`, `Integer.MAX_VALUE` and `Integer.MIN_VALUE`

**Comment by ankitson** [comment]
@xavier-fernandez Thanks for the catch, i added MAX & MIN - 0 was already in there. 

I also found and patched what I think is a bug in Integer.equals .. (int1).equals(int2) compares the result of downcasting int1 to a byte with int2, so it would fail when comparing numbers >= 256.

**Comment by xavier-fernandez** [comment]
@ankitson :)

LGTM

**Comment by densh** [comment]
LGTM, please squash in one commit and it's good to merge!

**Comment by ankitson** [comment]
squashed!

**Comment by ankitson** [comment]
squashed!

**Comment by densh** [comment]
Thanks @ankitson ! 👍

**Comment by andreaTP** [comment]
@ankitson I was on #242 too, are you still on it?

**Comment by ankitson** [comment]
@andreaTP I haven't really started on that besides this PR - if you are already working on it then please continue :)
